designs/PackageReference-Extern-Alias.md

```diff
@@ -0,0 +1,171 @@
+* Nomination updates on (project-system)[https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/Dependencies/PackageReference.xaml] side.
+
+## Future Work
+
+* Package metadata for transitive package references will be considered in the future. For now the recommendation is to elevate that PackageReference to a direct dependency. 
+
+## Open Questions
+
+* Will create follow ups for the partner issues once this design has been reviewed.
+* In which version will this functionality be ready?
+
+## Considerations
+
+In addition to the proposed approach, 2 of other solutions were considered. 
+
+### Do nothing - Recommend the customers use the custom target workaround
+
+The workaround that allows per assembly granularity is the following:
+
+```xml
+  <Target Name="AddCustomAliases" BeforeTargets="FindReferenceAssembliesForReferences;ResolveReferences">
+    <ItemGroup>
+      <ReferencePath Condition="'%(FileName)' == '$AssembleFileName$' AND '%(ReferencePath.NuGetPackageId)' == '$PackageId$'">
+        <Aliases>$Alias$</Aliases>
+      </ReferencePath>
+    </ItemGroup>
+  </Target>
+```
+
+Pros:
+
+* A very powerful solution, that covers both direct and transitive dependencies.
+* Allows per file granularity, if multiple assemblies are brought in by a package, each can have(or not) their own alias.
+
+Cons:
+
+* Important enough feature that deserves an out of the box solution.
+* The workaround recommends taking a dependency on tasks and targets that are best not depended on by customers. Goes against the policy of minimizing the size of the project files.
+* Not obvious, unnecessarily exposes the customers to restore/build internals.  
+
+### Support `Aliases` as a PackageReference property & provide a way to address transitively dependency out of the box
+
+A future evolution of the current approach could be the addition of metadata for transitive packages. 
+
+```xml
+<ItemGroup>
+    <PackageReference Include="StackExchange.Redis.StrongName.Signed" Version="1.26.0" Aliases="signed" />
+</ItemGroup>
+```
+
+For transitive packages we could introduce a new item type, where customers could provide the transitive package alias/NoWarn at some point in the future. 
+
+```xml
+<ItemGroup>
+    <TransitivePackageReference Include="StackExchange.Redis.StrongName.Signed" Aliases="signed" NoWarn="NU1701" />
+</ItemGroup>
+```
+
+This approach drastically increases the scope of this work, but given that it's additive. 
+Given that we are not confident that this is how we want to address the transitive package metadata problems this is just listed here for completeness.
+
+### References
+
+* https://github.com/NuGet/Home/issues/4989
+* https://github.com/NuGet/Home/issues/5740
+* https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/extern-alias
+* https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.projectreference.aliases?view=roslyn-dotnet

```

Comment:
Can you add descriptions/titles for these links?

Reply:
Sure, I can do that.